notes.md

```diff
@@ -489,81 +489,96 @@ an `Element` is calculated by subtracting the position of the `Element Data` of
+it easier for a `Matroska Player` to know which `Segments` are meant to be played together.
+
+The `SegmentFilename`, `PrevFilename` and `NextFilename` elements **MAY** also give hints on
+the original filenames that were used when the Segment links were created, in case some Segment UIDs are damaged.
 
 ## Hard Linking
 
-Hard Linking (also called splitting) is the process of creating a `Linked Segment`
-by relating multiple `Segment Elements` using the `NextUID` and `PrevUID` Elements.
-Within a `Linked Segment`, the timestamps of each `Segment` **MUST** follow consecutively
-in linking order.
-With Hard Linking, the chapters of any `Segment` within the `Linked Segment` **MUST**
-only reference the current `Segment`. With Hard Linking, the `NextUID` and `PrevUID` **MUST**
-reference the respective `SegmentUID` values of the next and previous `Segments`.
-The first `Segment` of a `Linked Segment` **SHOULD** have a `NextUID Element` and **MUST NOT**
-have a `PrevUID Element`.
-The last `Segment` of a `Linked Segment` **SHOULD** have a `PrevUID Element` and **MUST NOT**
-have a `NextUID Element`.
-The middle `Segments` of a `Linked Segment` **SHOULD** have both a `NextUID Element`
-and a `PrevUID Element`.
-
-In a chain of `Linked Segments` the `NextUID` always takes precedence over the `PrevUID`.
-So if SegmentA has a NextUID to SegmentB and SegmentB has a PrevUID to SegmentC,
-the link to use is SegmentA to SegmentB.
-If SegmentB has a PrevUID to SegmentA but SegmentA has no NextUID, then the Matroska Player
+Hard Linking, also called splitting, is the process of creating a `Linked Segment`
+by linking multiple `Segment Elements` using the `NextUID` and `PrevUID` Elements.
+
+All `Segments` within a `Hard Linked Segment` **MUST** use the same `Tracks` list and `TimestampScale`.
+
+Within a `Linked Segment`, the timestamps of `Block` and `SimpleBlock` **MUST** follow consecutively
+the timestamps of `Block` and `SimpleBlock` from the previous `Segment` in linking order.
+
+With Hard Linking, the chapters of any `Segment` within the `Linked Segment` **MUST** only reference the current `Segment`.
+The `NextUID` and `PrevUID` reference the respective `SegmentUID` values of the next and previous `Segments`.
+
+The first `Segment` of a `Linked Segment` **MUST NOT** have a `PrevUID Element`.
+The last `Segment` of a `Linked Segment` **MUST NOT** have a `NextUID Element`.
+
+For each node of the chain of `Segments` of a `Linked Segment` at least one `Segment` **MUST** reference the other `Segment` of the node.
+
+In a chain of `Segments` of a `Linked Segment` the `NextUID` always takes precedence over the `PrevUID`.
+So if SegmentA has a `NextUID` to SegmentB and SegmentB has a `PrevUID` to SegmentC,
+the link to use is `NextUID`.
+If SegmentB has a `PrevUID` to SegmentA but SegmentA has no `NextUID`, then the Matroska Player
 **MAY** consider these two Segments linked as SegmentA followed by SegmentB.
```

Comment:
This is not fully correct.

Hard-Linking is alwasy a question of "which Segment was started/used" by a user.

When a user starts SegmentA than is there no Hard-Linking because there is no `NextUID`.
Only if the user starts SegmentB than the player has to use Hard-Linking.